### PR TITLE
Update crate versions for 1.2 release

### PR DESCRIPTION
**Pull request by sffc**
Part of #3297

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]

**Review by Manishearth — APPROVED** [review]

**Comment by Manishearth** [comment]
Windows test seems to be consistently failing with a PDB link error

```
  = note: LINK : fatal error LNK1318: Unexpected PDB error; FILE_SYSTEM (3) 'D:\a\icu4x\icu4x\target\debug\deps\icu4x_verify_zero_copy-226492d5bfe2e4e7.pdb'
```

it's just verify-zero-copy though. Might have to do with the dhat crate? It's linux-only

**Comment by sffc** [comment]
Just above that message, it also says

```
2023-04-14T01:53:13.2182331Z ##[warning]You are running out of disk space. The runner will stop working when the machine runs out of disk space. Free space left: 57 MB
```

**Comment by sffc** [comment]
The warning about disk space is only visible when downloading the raw logs.

**Comment by robertbastian** [comment]
It baffles me how this PR can possibly break something. The only changes in the lock files are our version numbers.

File: utils/tinystr/Cargo.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ criterion = "0.4"
 postcard = { version = "1.0.0", features = ["use-std"], default-features = false }
 rand = { version = "0.8.5", features = ["small_rng"] }
 serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
-tinystr_old = { version = "0.4", package = "tinystr", features = ["serde"] }
+tinystr_old = { version = "0.7.1", package = "tinystr", features = ["serde"] }
```
Comment (Manishearth) [inline utils/tinystr/Cargo.toml:44]: this shouldn't happen; but also I think we can safely remove this test now

Comment (Manishearth) [reply utils/tinystr/Cargo.toml:44]: https://github.com/unicode-org/icu4x/pull/3326

This is the one "not just a version bump" lockfile change, perhaps once we undo that the windows build will work again

(I think it's possible we're quite close to a threshold and the minor lockfile change brings us over)

**Comment by Manishearth** [comment]
Moving to https://github.com/unicode-org/icu4x/pull/3327

**Pull request closed without merging**